src/Store/CommandStoreConnector.re

```diff
@@ -314,6 +331,42 @@ let start = (getState, contributedCommands) => {
     ("window.moveRight", state => windowMoveEffect(state, Right)),
     ("window.moveUp", state => windowMoveEffect(state, Up)),
     ("window.moveDown", state => windowMoveEffect(state, Down)),
+    (
+      "terminal.new.vertical",
+      ({terminals, _}) => {
+        let nextTerminalId = Feature_Terminal.getNextId(terminals);
+        let bufferName = Feature_Terminal.getBufferName(nextTerminalId);
+        let cmd = Feature_Terminal.shellCmd;
+        multiActionEffect([
+          Actions.OpenFileByPath(
+            bufferName,
+            Some(WindowTree.Vertical),
+            None,
+          ),
+          Actions.Terminal(
+            Feature_Terminal.Started({id: nextTerminalId, cmd}),
+          ),
+        ]);
+      },
```

Comment:
Thanks for the detailed response, @glennsl ! This is very helpful. I knew it wasn't ideal, but wasn't sure how to address it - gave me some good ideas.

> Ideally it wouldn't know anything about it, and would just contribute commands like terminal.new, which when triggered could be passed to update with CommandTriggered("terminal.new), and would return an outMsg TerminalCreated(...).

I do like the idea of features contributing commands. We don't have any infrastructure for this yet, do we? It sounds like it could be cleaner than our giant `CommandStoreConnector` block.

> Either dispatch a msg to have Feature_Terminal.update update the state with a new terminal and return an outMsg telling its parent to open the buffer. I've used an outMsg in Feature_SCM for example, but the msg in this case would be little more than a function call.

This sounds like a great improvement. It helps keep more of the concerns inside `Feature_Terminal` - I was able to remove the exposing of the `id` and other internals with this approach.

I created an `outmsg` with `OpenBuffer` and `Nothing` as the 'effects'.

I also changed the `Feature_Terminal.Started` message to `Feature_Terminal.NewTerminal` - `Started` seems inaccurate, because as you mentioned, it hasn't actually started yet. `NewTerminal` sounds like a function call, but is at least more accurate than `Started`. I think the ideal would be to connect directly to the `CommandTriggered` as you suggest, but we don't have that scaffolding yet - let me know if you have naming suggestions in the meantimes.

> My first proposal is probably closest top the ideal while still being relatively easy to implement. And I think we can live with the function call message for a little bit.

Sounds good!

Reply:
> I do like the idea of features contributing commands. We don't have any infrastructure for this yet, do we?

Not as far as I'm aware. I think the command infrastructure could use an overhaul in general. It violates single-source-of-truth wrt. contributed commands, hides the command spec so they can't be looked up by id, and isn't set up to get updated when new extensions are added.

> I created an outmsg with OpenBuffer and Nothing as the 'effects'.

`TerminalCreated` would be more flexible and event-like, and the feature shouldn't need to know about the existence of buffers.

> I also changed the Feature_Terminal.Started message to Feature_Terminal.NewTerminal -

:+1: That seems good enough for now.

Reply:
>  I think the command infrastructure could use an overhaul in general.

Definitely!